packages/nvidia_nat_test/src/nat/test/llm.py

```diff
@@ -0,0 +1,207 @@
+            return AIMessage(content=chooser.next_response())
+
+        def stream(self, messages: Any, **_kwargs: Any) -> Iterator[AIMessage]:
+            chooser.sync_sleep()
+            yield AIMessage(content=chooser.next_response())
+
+        async def astream(self, messages: Any, **_kwargs: Any) -> AsyncGenerator[AIMessage, None]:
+            await chooser.async_sleep()
+            yield AIMessage(content=chooser.next_response())
+
+    yield LangChainTestLLM()
+
+
+@register_llm_client(config_type=TestLLMConfig, wrapper_type=LLMFrameworkEnum.LLAMA_INDEX)
+async def test_llm_llama_index(config: TestLLMConfig, builder: Builder):
+    """LLM client for LlamaIndex"""
+    del builder  # suppress linting error
+    chooser = _ResponseChooser(response_seq=config.response_seq, delay_ms=config.delay_ms)
+
+    class LIChatMessage:
+
+        def __init__(self, content: str) -> None:
+            self.content = content
+
+    class LIChatResponse:
+
+        def __init__(self, text: str) -> None:
+            self.message = LIChatMessage(text)
+            self.text = text
+
+    class LITestLLM:
+
+        def chat(self, messages: list[Any] | None = None, **_kwargs: Any) -> LIChatResponse:
+            chooser.sync_sleep()
+            return LIChatResponse(chooser.next_response())
+
+        async def achat(self, messages: list[Any] | None = None, **_kwargs: Any) -> LIChatResponse:
+            await chooser.async_sleep()
+            return LIChatResponse(chooser.next_response())
+
+        def stream_chat(self, messages: list[Any] | None = None, **_kwargs: Any) -> Iterator[LIChatResponse]:
+            chooser.sync_sleep()
+            yield LIChatResponse(chooser.next_response())
+
+        async def astream_chat(self,
+                               messages: list[Any] | None = None,
+                               **_kwargs: Any) -> AsyncGenerator[LIChatResponse, None]:
+            await chooser.async_sleep()
+            yield LIChatResponse(chooser.next_response())
+
+    yield LITestLLM()
+
+
+@register_llm_client(config_type=TestLLMConfig, wrapper_type=LLMFrameworkEnum.CREWAI)
+async def test_llm_crewai(config: TestLLMConfig, builder: Builder):
+    """LLM client for CrewAI"""
+    del builder  # suppress linting error
+    chooser = _ResponseChooser(response_seq=config.response_seq, delay_ms=config.delay_ms)
+
+    class CrewAITestLLM:
+
+        def call(self, messages: list[dict[str, str]] | None = None, **kwargs: Any) -> str:
+            chooser.sync_sleep()
+            return chooser.next_response()
+
+    yield CrewAITestLLM()
+
+
+@register_llm_client(config_type=TestLLMConfig, wrapper_type=LLMFrameworkEnum.SEMANTIC_KERNEL)
+async def test_llm_semantic_kernel(config: TestLLMConfig, builder: Builder):
+    """LLM client for SemanticKernel"""
+    from semantic_kernel.contents.chat_message_content import ChatMessageContent
+    from semantic_kernel.contents.utils.author_role import AuthorRole
```

Comment:
Given that we depend on a `semantic_kernel` imports, `nvidia-nat-test` needs `nvidia-nat[semantic-kernel]` as a dependency